### PR TITLE
Wip/hughsie/cpuid

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: plugins/cpu/fu-cpu-device.c [hunk 1]
```diff
@@ -153,34 +153,61 @@ fu_cpu_device_init (FuCpuDevice *self)
...
 	fu_device_add_instance_id (device, devid3);
 	return TRUE;
 }
 
+static gboolean
+fu_cpu_device_probe (FuDevice *device, GError **error)
+{
+	if (!fu_cpu_device_add_instance_ids (device, error))
+		return FALSE;
+	return TRUE;
```
Comment (superm1) [inline plugins/cpu/fu-cpu-device.c:243]: maybe just `return fu_cpu_device_add_instance_ids();`?

File: plugins/cpu/fu-cpu-device.c [hunk 2]
```diff
@@ -35,6 +35,64 @@ fu_cpu_device_to_string (FuDevice *device, guint idt, GString *str)
 				    fu_cpu_device_has_flag (self, FU_CPU_DEVICE_FLAG_SMAP));
 }
 
+static const gchar *
+fu_cpu_device_convert_vendor (const gchar *vendor)
```
Comment (superm1) [inline plugins/cpu/fu-cpu-device.c:39]: Were these sourced from kernel?  Might be good to reference in a comment where to get updates from

File: plugins/cpu/fu-cpu-device.c [hunk 3]
```diff
@@ -92,37 +100,149 @@ fu_cpu_device_init (FuCpuDevice *self)
 	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_INTERNAL);
 	fu_device_add_icon (FU_DEVICE (self), "computer");
 	fu_device_set_version_format (FU_DEVICE (self), FWUPD_VERSION_FORMAT_HEX);
+	fu_device_set_physical_id (FU_DEVICE (self), "cpu:0");
```
Comment (superm1) [inline plugins/cpu/fu-cpu-device.c:103]: This shouldn't be hardcoded in case of a multi-socket system no?

File: plugins/cpu/fu-cpu-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/cpu/fu-cpu-device.c:243]: I know it looks janky at the moment, but the later patches insert other things in here too.

File: plugins/cpu/fu-cpu-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/cpu/fu-cpu-device.c:39]: Wikipedia! Is that good enough?

File: plugins/cpu/fu-cpu-device.c (hunk 3 above)
Comment (hughsie) [reply plugins/cpu/fu-cpu-device.c:103]: I'm explicitly just querying the first physical processor;I don't think it's super interesting to enumerate all of them like before.

File: plugins/cpu/fu-cpu-device.c (hunk 2 above)
Comment (superm1) [reply plugins/cpu/fu-cpu-device.c:39]: Probably..

**Pull request merged**